resources/assets/coffee/react/comments-show/main.coffee

```diff
@@ -0,0 +1,118 @@
+###
+#    Copyright 2015-2018 ppy Pty. Ltd.
+#
+#    This file is part of osu!web. osu!web is distributed with the hope of
+#    attracting more community contributions to the core ecosystem of osu!.
+#
+#    osu!web is free software: you can redistribute it and/or modify
+#    it under the terms of the Affero GNU General Public License version 3
+#    as published by the Free Software Foundation.
+#
+#    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+{a, button, div, h1, li, ol, p, span} = ReactDOMFactories
+el = React.createElement
+
+class CommentsShow.Main extends React.PureComponent
+  constructor: (props) ->
+    super props
+
+    @id = "comments-show-#{osu.uuid()}"
+
+    comments = osu.updateCollection @props.data.bundle.comments, [@props.data.comment]
+    users = osu.updateCollection @props.data.bundle.users, [@props.data.comment.user, @props.data.comment.editor]
+
+    @state =
+      comments: comments
+      users: users
+
+
+  componentDidMount: =>
+    $.subscribe "comments:added.#{@id}", @appendBundle
+    $.subscribe "comment:updated.#{@id}", @update
+
+
+  componentWillUnmount: =>
+    $.unsubscribe ".#{@id}"
+
+
+  render: =>
+    commentsByParentId = _(@state.comments)
+      .uniqBy('id')
+      .orderBy(['created_at', 'id'], ['desc', 'desc'])
+      .groupBy('parent_id')
+      .value()
+    usersById = _.keyBy(@state.users ? [], 'id')
+
+    mainComment = commentsByParentId[@props.data.comment.parent_id][0]
```

Comment:
is there a case where this isn't just the current comment?

Reply:
when the current comment is edited/deleted/restored.